webfront/views/modifiers.py

```diff
@@ -260,19 +260,26 @@ def filter_by_key_species(value, general_handler):
 
 def filter_by_entry(value, general_handler):
     queryset = general_handler.queryset_manager.get_queryset()
-    response = TaxonomyPerEntry.objects.filter(taxonomy__in=queryset).filter(
-        entry_acc__accession__iexact=value
-    )
+    response = (TaxonomyPerEntry.objects
+                .filter(taxonomy__in=queryset)
+                .filter(entry_acc__accession=value.upper())
+                )
     if len(response) == 0:
-        raise EmptyQuerysetError("No documents found with the current selection")
+        response = (TaxonomyPerEntry.objects
+                    .filter(taxonomy__in=queryset)
+                    .filter(entry_acc__accession=value.lower())
+                    )
+        if len(response) == 0:
+            raise EmptyQuerysetError("No documents found with the current selection")
     return response.first()
```

Comment:
1. Considering that the `filter_by_entry` modifier is only available when querying a specific taxon, can't we use `taxonomy__accession=queryset[0].accession` instead of `taxonomy__in=queryset`?
2. Instead of converting the accession to upper case, then lower case, can't we use `entry_acc__accession__iexact=value`?

```py
def filter_by_entry(value, general_handler):
    queryset = general_handler.queryset_manager.get_queryset()
    if len(queryset) == 0:
        raise EmptyQuerysetError("No documents found with the current selection")

    response = (TaxonomyPerEntry.objects
                .filter(taxonomy__accession=queryset[0].accession)
                .filter(entry_acc__accession__iexact=value)
                )

    if len(response) == 0:
        raise EmptyQuerysetError("No documents found with the current selection")

    return response.first()
```

Reply:
1. Yeah, I was in 2 minds about that one. I think, technically the modifier could be used in URLs of the type `[endpoint/DB]/[endpoint/DB]`. I mean it's not currently used but if we put it there then the query using `in` will still work, that however gets to waste once we use `.first()` in the return, so yeah probably worth changing. Anyways doing quick tests the `in` wasn't the problem stopping the use of the index.
2. The problem was `__iexact`. We were using it before, and I never noticed but because of it the index of the table was not used and those queries were too slow. Making navigating the taxonomy tree quite a pain. I know the code looks dumb and repetitive, but is significantly faster. It clearly doesn't cover all the cases that `__iexact` and only checks all upper or all lower, but given that we are checking accessions, I think it is safe to use.